src/ansys/fluent/core/workflow.py

```diff
@@ -902,10 +902,15 @@ def __setattr__(self, attr, value):
             self.set_state({attr: value})
 
     def __dir__(self):
-        arg_list = []
-        for arg in self():
-            arg_list.append(camel_to_snake_case(arg))
-        return sorted(set(list(self.__dict__.keys()) + dir(type(self)) + arg_list))
+        arg_list = (
+            [arg for arg in self() if isinstance(self(), dict)]
```

Comment:
Calling `get_state()` is much more readable here.

Reply:
Done, Thank you.